sdks/java/core/src/test/java/org/apache/beam/sdk/coders/org/apache/beam/sdk/coders/RowCoderTest.java

```diff
@@ -145,4 +149,47 @@ public void testVerifyDeterministicNestedRow() throws NonDeterministicException
 
     coder.verifyDeterministic();
   }
+
+  @Test
+  public void testConsistentWithEqualsBytesField() {
+    Schema schema = Schema.of(Schema.Field.of("f1", FieldType.BYTES));
+    Row row = Row.withSchema(schema).addValue(new byte[] {1, 2, 3, 4}).build();
+    RowCoder coder = RowCoder.of(schema);
+
+    CoderTestUtils.testConsistentWithEquals(coder, row);
+  }
+
+  @Test
+  public void testStructuralValueBytesField() {
+    Schema schema = Schema.of(Schema.Field.of("f1", FieldType.BYTES));
+    Row row = Row.withSchema(schema).addValue(new byte[] {1, 2, 3, 4}).build();
+    RowCoder coder = RowCoder.of(schema);
+
+    CoderTestUtils.testStructuralValueConsistentWithEquals(coder, row);
+  }
+
+  @Test
+  @Ignore // TODO should it even be possible to create rows with such schema?
```

Comment:
Need advice what to do

Reply:
This seems like an OK schema to me. What is the problem with it?

Reply:
`Map` implementations imply that there is either order or hashCode+equality for values. `byte[]`doesn't have either of that. `Map<byte[], ?>` will work only if we provide specialized implementation for it.